tests/test_secret_tools.py

```diff
@@ -240,3 +243,115 @@ def test_decrypt_secret_environment_for_service(mock_get_secret_provider):
         {"SECRET": "SECRET(123)"}
     )
     assert ret == mock_secret_provider.decrypt_environment.return_value
+
+
+@mock.patch("paasta_tools.secret_tools.get_secret_provider", autospec=True)
+def test_decrypt_secret_volumes_multiple_files(mock_get_secret_provider):
+    mock_secret_provider = mock.Mock()
+    mock_get_secret_provider.return_value = mock_secret_provider
+
+    mock_secret_volumes_config = [
+        SecretVolume(
+            container_path="/the/container/path/",
+            items=[
+                {"key": "the_secret_name1", "path": "the_secret_filename1"},
+                {"key": "the_secret_name2", "path": "the_secret_filename2"},
+            ],
+        )
+    ]
+    mock_secret_provider.decrypt_secret_raw.side_effect = [
+        "the_secret_contents1",
+        "the_secret_contents2",
+    ]
+    ret = decrypt_secret_volumes(
+        secret_provider_name="vault",
+        secret_volumes_config=mock_secret_volumes_config,
+        soa_dir="/nail/blah",
+        service_name="universe",
+        cluster_name="mesosstage",
+        secret_provider_kwargs={"some": "config"},
+    )
+    assert ret == {
+        "/the/container/path/the_secret_filename1": "the_secret_contents1",
+        "/the/container/path/the_secret_filename2": "the_secret_contents2",
+    }
+
+    assert mock_secret_provider.decrypt_secret_raw.call_args_list == [
+        mock.call(
+            "the_secret_name1",
+        ),
+        mock.call(
+            "the_secret_name2",
+        ),
+    ]
```

Comment:
hmm, do we need this test? seems like this just tells us that we called some function and passed its args in the right order - not that the expected behavior occurred :)

Reply:
Fair, but this test was largely based on the methodology in extant test above it for testing decrypt_secret_environment. 

It exercises the code paths I added, and confirms the logic that merges the `items` names with the container path is correct. 

Edit: Oh, is your point specifically about this assert or the test as a whole? I think you're right that the function parameters are not a very useful assertion, and I can remove them.

Reply:
Applying the same changes for the similar tests in `test_kubernetes_tools`

Reply:
oh 🤦🏻 - english is hard: i just meant the call_args_list assert